server/app/controllers/files.js

```diff
@@ -0,0 +1,38 @@
+const uuid = require('uuid/v4');
+const AWS = require('aws-sdk');
+const asyncHandler = require('../middleware/async');
+const ErrorResponse = require('../middleware/error');
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.AWS_ACCESS_KEY,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+});
+
+const { BUCKET_NAME, UPLOAD_DIRNAME } = process.env;
+
+const upload = asyncHandler(async (req, res, next) => {
+  if (!req.files || !req.files.file) {
+    next(new ErrorResponse('File is undefined', 400));
+    return;
```

Comment:
Having the ``return`` below the ``next()`` function means that the upload function will pass the ErrorResponse to the next() function and then return back to the upload function, which then ends. There is too much jumping around in terms of the function calls. So put the ``return`` next to the ``next()`` call

``
return next(new ErrorResponse('File is undefined', 400));
``

Reply:
I agree that having return on the same line makes more sense. The only reason I put it on the next line is because I would get a linter warning about arrow function not returning anything. That warning doesn't always come up for every `return next(new ErrorResponse( ...`, so I'm not sure why. But we can change it later.

Reply:
Turned out that I was missing a return statement at the end of the block, which was causing the problem. I've added the change as you commented. Thanks!